### PR TITLE
feat: introduce @ngrx/operators package

### LINKED ISSUE #4057: Add `@ngrx/operators` package
### Which @ngrx/* package(s) are relevant/related to the feature request?

component-store, effects, signals

### Information

Create the `@ngrx/operators` package that will contain the following operators:

- `concatLatestFrom` from `@ngrx/effects` that can be also used with `@ngrx/component-store` and/or upcoming `@ngrx/signals` package.
- `tapResponse` from `@ngrx/component-store` that can be used with `@ngrx/signals`.

### Describe any alternatives/workarounds you're currently using

_No response_

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [ ] No

### PR DESCRIPTION
**Pull request by brandonroberts**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/main/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Closes #4057 

## What is the new behavior?

Introduces new `@ngrx/operators` package for the following operators to be shared across NgRx packages

- `tapResponse` from `@ngrx/component-store`
- `concatLatestFrom` from `@ngrx/effects`

The `@ngrx/operators` package will be added to these packages as a dependency, and the operators will be re-exported to prevent any breaking changes.

The `@ngrx/operators` package can also be installed as a standalone dependency.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by timdeschryver — APPROVED** [review]
🚀 Thanks Brandon!
Should `operators` also be added to `publish-latest` and `publish-next`?

**Comment by brandonroberts** [comment]
> 🚀 Thanks Brandon! Should `operators` also be added to `publish-latest` and `publish-next`?

Good catch. Added `signals` and `operators`

File: modules/operators/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      "@ngrx/store",
+      "@ngrx/effects",
+      "@ngrx/entity",
+      "@ngrx/router-store",
+      "@ngrx/data",
+      "@ngrx/schematics",
+      "@ngrx/store-devtools",
+      "@ngrx/component-store",
+      "@ngrx/component",
+      "@ngrx/eslint-plugin"
```
Comment (markostanimirovic) [inline modules/operators/package.json:38]:
```suggestion
      "@ngrx/eslint-plugin",
      "@ngrx/signals"
```

We can copy this list to each `modules/**/package.json`

File: modules/operators/public_api.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export * from './src/index';
```
Comment (markostanimirovic) [inline modules/operators/public_api.ts:1]: Is the `public_api.ts` file still needed? `tsconfig.build.json` and `ng-package.json` from `@ngrx/operators` already point to `index.ts`. (Btw, I removed the `public_api` file from `@ngrx/signals`)

File: build/publish-latest.ts [hunk 3]
```diff
@@ -17,6 +17,8 @@ export async function publishLatestToNpm() {
     'schematics',
     'eslint-plugin',
     'data',
+    'signals',
+    'operators',
```
Comment (markostanimirovic) [inline build/publish-latest.ts:21]: We can now rollback changes from [`config.ts`](https://github.com/ngrx/platform/pull/4002/files#diff-346737222179600781b6e608dd452f1f9e662081473dc1d0ad51bb6bf21250dbR24) and [`copy-schematics-core.ts`](https://github.com/ngrx/platform/pull/4002/files#diff-ed45f28768886664e000958eafce01133876762a7f7fb8d0a88f6cfe74e389b8R12)

**Review by markostanimirovic — COMMENTED** [review]
Great work Brandon! 🥇

**Pull request merged**